tests/e2e/kcs/test_selinux_relabel_solution.py

```diff
@@ -330,3 +355,102 @@ def test_selinux_relabel_for_existing_pvc(
         assert (
             pod_restart_time_before_fix > pod_restart_time_after_fix
         ), "Time taken for pod restart after fix is more than before fix."
+
+    @polarion_id("OCS-5163")
+    @pytest.mark.parametrize("copies", [5])
+    def test_selinux_relabel_for_new_pvc(
+        self,
+        pvc_factory,
+        service_account_factory,
+        storageclass_factory,
+        copies,
+        teardown_factory,
+    ):
+        """
+        Steps:
+            1. Create storageclass with security context set.
+            2. Create cephfs pvcs and attach pod with more than 100K files across multiple nested directories
+            3. Take md5sum for them some random files and get pod restart time
+            4. Restart the pod which is hosting cephfs files in large numbers.
+            5. Check data integrity.
+            6. Check for relabeling - this should not be happening.
+
+        Args:
+            pvc_factory (function): A call to pvc_factory function
+            service_account_factory (function): A call to service_account_factory function
+            storageclass_factory ((function): A call to storageclass_factory function
+            copies (int): number of copies to write kernel files in pod
+
+        """
+        # Create storageclass with security context
+        self.ocp_project = ocp.OCP(
+            kind=constants.NAMESPACE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        sc_name = "ocs-storagecluster-cephfs-selinux-relabel"
+        self.storage_class = storageclass_factory(
+            sc_name=sc_name,
+            interface=constants.CEPHFILESYSTEM,
+            kernelMountOptions='context="system_u:object_r:container_file_t:s0"',
+        )
+
+        # Create PVC using new storageclass
+        self.pvc_obj = pvc_factory(
+            project=self.ocp_project,
+            storageclass=self.storage_class,
+            size="20",
+        )
+        log.info(f"PVC {self.pvc_obj.name} created")
+        teardown_factory(self.pvc_obj)
```

Comment:
you are using pvc_factory for creating PVCs, do you still need to add these PVC objects to the teardown factory here?